polars/polars-core/src/cloud.rs

```diff
@@ -114,9 +114,14 @@ impl CloudOptions {
             .aws
             .as_ref()
             .ok_or_else(|| polars_err!(ComputeError: "`aws` configuration missing"))?;
-        AmazonS3Builder::new()
-            .try_with_options(options.clone().into_iter())
-            .and_then(|b| b.with_bucket_name(bucket_name).build())
+
+        let mut builder = AmazonS3Builder::new();
+        for (key, value) in options.iter() {
+            builder = builder.with_config(*key, value);
+        }
```

Comment:
Feels like there must be some snappy Rust syntax to do this with less code... this works, though.